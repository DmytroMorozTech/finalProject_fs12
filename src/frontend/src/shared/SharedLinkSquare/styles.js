import {makeStyles} from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  link: {
    display: 'flex',
    alignItems: 'center',
    cursor: 'pointer',
    fontSize: theme.typography.h5.fontSize,
    paddingBottom: theme.spacing(1),
    padding: theme.spacing(2),
    textDecoration: 'none',
    color: theme.palette.grey[700],
    '&:hover': {
      borderRadius: theme.shape.small,
      backgroundColor: theme.palette.grey[200]
    }
  }
}))
