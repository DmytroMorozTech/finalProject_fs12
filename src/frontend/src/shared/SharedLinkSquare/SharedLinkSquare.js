import styles from './styles'
import clsx from 'clsx'
import { Link } from 'react-router-dom'

function SharedLinkSquare ({ title, to = '#', icon, className = '', onClick }) {
  const classes = styles()

  return (
    <Link to={to} className={clsx(classes.link, className)}>
      <div onClick={onClick}>
        {title}
        {icon}
      </div>
    </Link>
  )
}

export default SharedLinkSquare
