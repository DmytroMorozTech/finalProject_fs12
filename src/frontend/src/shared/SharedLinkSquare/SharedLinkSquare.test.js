import React from 'react'
import SharedLinkSquare from './SharedLinkSquare'
import { shallow } from 'enzyme'

const setUp = () => shallow(<SharedLinkSquare />)

describe('should render SharedLinkSquare component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
