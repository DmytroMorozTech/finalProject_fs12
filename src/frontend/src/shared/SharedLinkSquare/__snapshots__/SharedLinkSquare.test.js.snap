// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SharedLinkSquare component to match snapshot 1`] = `
<Link
  className="makeStyles-link-1"
  to="#"
>
  <div />
</Link>
`;
